operator/endpointslicegc/gc.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package endpointslicegc
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"time"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cilium_api_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+// params contains all the dependencies for the endpoint-slice-gc.
+// They will be provided through dependency injection.
+type params struct {
+	cell.In
+
+	Logger    *slog.Logger
+	Lifecycle cell.Lifecycle
+	JobGroup  job.Group
+
+	Clientset            k8sClient.Clientset
+	CiliumEndpointSlices resource.Resource[*cilium_api_v2a1.CiliumEndpointSlice]
+
+	SharedCfg SharedConfig
```

Comment:
Nit. Could depend on `ciliumendpointslices.SharedConfig` instead of defining a new one, as they'll eventually need to converge anyways.